include/common/aglResCommon.h

```diff
@@ -0,0 +1,198 @@
+#pragma once
+
+#include <basis/seadRawPrint.h>
+#include <prim/seadBitUtil.h>
+
+namespace agl {
+
+// maybe first parameter is is_le, maybe it is big_endian - different between decomps
+void ModifyEndianU32(bool is_le, void* p_data, size_t size);
+
+template <typename _DataType>
```

Comment:
```suggestion
template <typename DataTypeT>
```
`_DataType` is a reserved identifier (because it starts with an underscore and an uppercase letter) and should be renamed

Reply:
I guess `DataType_` would be better then.

Reply:
Which name would be the better choice here - `DataTypeT` or `DataType_`?

Reply:
imo `DataType_`.
`DataTypeT` = "Data Type Type", it's like saying "smh my head".

Reply:
Alright, changed it accordingly. If someone else has a different opinion, we can discuss about it again.